### PR TITLE
feat: add `join` and `joinRaw`

### PR DESCRIPTION
**Pull request by kyranet**
A little utility `lexure` had that we didn't. Most useful for https://github.com/sapphiredev/framework/pull/531.

### CODE REVIEW

**Review by vladfrangu — APPROVED** [review]

File: packages/lexure/src/lib/util/util.ts [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+ */
+export function joinRaw(parameters: readonly Parameter[]) {
+	if (parameters.length === 0) return '';
+	if (parameters.length === 1) return parameters[0].raw;
+
+	let output = parameters[0].raw;
+	for (let i = 1; i < parameters.length; i++) {
+		const parameter = parameters[i];
+		output += parameter.leading + parameter.raw;
+	}
```
Comment (favna) [inline packages/lexure/src/lib/util/util.ts:36]:
```suggestion
	for (const parameter of parameters) {
		output += parameter.leading + parameter.raw;
	}
```

Were you half asleep or something

File: packages/lexure/src/lib/util/util.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+ */
+export function join(parameters: readonly Parameter[]) {
+	if (parameters.length === 0) return '';
+	if (parameters.length === 1) return parameters[0].value;
+
+	let output = parameters[0].value;
+	for (let i = 1; i < parameters.length; i++) {
+		const parameter = parameters[i];
+		output += parameter.leading + parameter.value;
+	}
```
Comment (favna) [inline packages/lexure/src/lib/util/util.ts:17]:
```suggestion
	for (const param of parameters) {
		output += param.leading + param.value;
	}
```

**Review by favna — CHANGES_REQUESTED** [review]

File: packages/lexure/src/lib/util/util.ts (hunk 1 above)
Comment (kyranet) [reply packages/lexure/src/lib/util/util.ts:36]: It starts from 1, not 0.

Comment (favna) [reply packages/lexure/src/lib/util/util.ts:36]: I guess I'm the one half asleep then

**Review by favna — APPROVED** [review]

**Pull request merged**